### PR TITLE
Post module item completion requirements

### PR DESCRIPTION
**Pull request by armstrongnate**
pork: Refactor Session.swift to not use URLSession delegate methods
and to use callbacks instead. This fixes some of the bad access errors
we were getting when accessing the stored completion handlers.

pork: use formSheet for most submission types (per Dave's request)

refs: MBL-12870
affects: student
release note: none

### CODE REVIEW

File: Canvas/CanvasUITests/Notifications/NotificationsListTests.swift [hunk 1]
```diff
@@ -8,6 +8,7 @@
 // License, or (at your option) any later version.
 //
 // This program is distributed in the hope that it will be useful,
+l
```
Comment (bluwave) [inline Canvas/CanvasUITests/Notifications/NotificationsListTests.swift]: ??

Comment (armstrongnate) [reply Canvas/CanvasUITests/Notifications/NotificationsListTests.swift]: What the. How come I'm not seeing that..

Comment (armstrongnate) [reply Canvas/CanvasUITests/Notifications/NotificationsListTests.swift]: Oh j/k I see it. Thanks!

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  This pull request will not generate a release note.
  </td>
    </tr>
  </tbody>
</table>


#### Affected Apps: Student

[MBL-12870](https://instructure.atlassian.net/browse/MBL-12870)

Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Core** | 95.34% | 95.33% | 0.01%
**Student** | 91.12% | 91.1% | 0.03%
**Teacher** | 100% | 100% | 0%
**React Native** | 92.79% | 92.79% | 0%
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Pull request merged**